design/disaster-recovery-snapshots.md

```diff
@@ -0,0 +1,128 @@
+# Optionally enable disaster recovery snapshots
+
+Currently KOTS supports application level snapshots, which can be used to restore a previously deployed application version along with its data.
+However in case of a disaster, only application can be restored to a new cluster and Admin Console can never be used to manage it again.
+This document proposes a way to implement disaster recovery snapshots using Velero.
+
+## Goals
+
+1. Use Velero to create Admin Console snapshot.
+1. Make CLI based disaster recovery possible.
+1. Propose a way to enable UI based recovery process.
+
+## Non Goals
+
+1. Improving application deployment process to avoid the `field is immutable` errors.
+1. Performing incremental backups of docker images.
```

Comment:
Maybe include a non-goal, for now, of backing up and restoring things outside of the admin console. e.g. for  kURL installation, there's a registry which Kots needs, but isn't backed up. At some point we're going to need to solve that, of course.

Reply:
Should we make it explicit that restoring individual apps is a non-goal?

If it *is* a goal, then the big challenge I have is that you restore admin console and all apps will show up interface, even if they're non-existent. 

Related, should we make it a requirement that when restoring the admin console, you get all apps? At most with a user guide for people that really know what they're doing if somehow admin console gets corrupt, but their app isn't and they don't want to restore the app. But certainly by default, restoring the admin console should do a full restore.